backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/NumaPolicyUnit.java

```diff
@@ -66,6 +74,14 @@ public NumaPolicyUnit(PolicyUnit policyUnit,
                 }
                 skipHostCheck = false;
 
+                if (VmCpuCountHelper.isAutoPinning(vm)) {
+                    boolean vmIsSet = initiateAutoPinToHost(vm, host);
```

Comment:
that's pretty ugly as i set here on each host the whole policy configuration (cpu pinning needed for NumaPinningHelper.findAssignment and for cpu pinning i need the cpu topology).

not only that, it will be executed twice: once on validation of runVm and then by selecting a host.

any thought to make it nicer?

Reply:
What is this code supposed to do? You've already filtered out the hosts that do not have the same amount of sockets and numa nodes.